### ISSUE TITLE
Feature Requests: Key Items, Stats, and Characters

### ISSUE THREAD

**Issue by nintendesi** [PheonixMMKC777/Final_Fantasy_2_Randomizer#1]
Thank you so much for all of your work with this randomizer. It has so much potential, and it seems to be an excellent work in progress. I noticed a few issues that you may be aware of, but in case you haven't noticed them, I wanted to call attention to them.

The first issue occurs when I select the option "Randomize + Fill Enemy Loot" in the program. With this option, key items (e.g. Snowcraft, Pendant, etc.) are randomized in the pool of enemy items. This is an interesting feature, but the problem is that these key items don't seem to function (at least the few that I was able to test). For example, I won the Pendant item from a battle, then visited the Wyvern in Castle Deist. The Wyvern, however, did not give me the Wyvern Egg. I had to visit Deist Cavern, collect the Pendant from the vanilla location (which gave me two Pendants, permanently filling up slots in my inventory), and then the Wyvern gave me the Wyvern Egg.

It would be very nice to see properly functioning key items that the player wins from battles. This would allow the player to skip visiting the vanilla location for the key item. Alternatively, key items could be removed from the enemy loot item pool. Better yet, it could be an option to place key items in the enemy loot item pool (assuming they function properly).

The second issue I experienced had to do with the option labeled "Stats Never Go Down." I enabled this option, but I found that my stats still decremented. I wanted to ensure that the stats actually decreased, so I am providing screenshots with timestamps to show that a character's stat decreased (Stamina).

Below is one of my character's stats before going into a battle.

![Final_Fantasy_2_Tr_2 2022-03-18 12 17 09](https://user-images.githubusercontent.com/76424048/159084573-4667d299-6319-4651-9f83-5d8eebc26107.png)

After the battle, this character received some stat changes, and the game informed me that Stamina went down.

![Final_Fantasy_2_Tr_2 2022-03-18 12 26 02](https://user-images.githubusercontent.com/76424048/159084574-9ea1b13a-e7eb-4fa7-ae9b-cbbedd20eb68.png)

In case the text was a minor bug, I went to verify this character's stats after the battle. Indeed, the character's Stamina decreased after the battle, despite that I enabled "Stats Never Go Down."

![Final_Fantasy_2_Tr_2 2022-03-18 12 30 01](https://user-images.githubusercontent.com/76424048/159084575-b2d945f8-5f23-4144-b211-2a035decd5bc.png)

Finally, this isn't so much an issue but rather a feature request. I was expecting to randomize the characters by shuffling them around, and I was surprised to see duplicate characters. A nice feature for the randomizer would be to have an option to prevent duplicate characters, but this is not as significant as addressing the earlier two issues above.

![Final_Fantasy_2_Tr_2 2022-03-18 16 20 35](https://user-images.githubusercontent.com/76424048/159084579-da17b425-b727-497d-8ec2-70290000bf2b.png)

Thank you again for all of your work on this randomizer, and I look forward to any more work that might be done on it in the future!

**Comment by PheonixMMKC777** [comment]
Yeah I am aware that key items like the wyvern egg do not trigger the events, I am not entirely sure how the game handles Story Progression, so for the time being I plan on removing them from the item pool.

Duplicate Characters Have been an issue and I have just recently fixed that, also along with it crashing while randomizing player characters.

Also mistake on my part on the Stats Never Go Down, now that I think about it, I only wrote code for disabling the attack stat going down, not all stats.  I will fix that soon

Also are you using the Executable or Powershell Version?

**Comment by nintendesi** [comment]
Thank you so much for all of your help and the prompt updates! I look forward to all future work on this project!

I am using the Executable for the randomizer.
